mempool/src/pool/mod.rs

```diff
@@ -968,9 +970,36 @@ impl<M: MemoryUsageEstimator> Mempool<M> {
             .expect("best index to exist");
         let tx_source = TransactionSourceForConnect::for_mempool(&best_index);
 
+        let mut unique_txids = BTreeSet::new();
+        let mut ordered_txids = vec![];
+
+        for transaction in tx_accumulator.transactions().iter() {
+            // Use transactions already in the Accumulator to sort for
+            // uniqueness only i.e don't send them through the
+            // verifier
+            unique_txids.insert(transaction.transaction().get_id());
+        }
```

Comment:
Also I don't think the case when a transaction given by hex spends the same utxo as one in mempool is handled properly.

It would be nice to start with a test to cover this:
* Add a transaction that spends an outpoint to mempool
* Construct a different transaction that spends the same outpoint, prepopulate the accumulator with it
* Call `collect_txs` with the accumulator and `fill_from_mempool` set `true`

Reply:
Yep, I found that this scenario failed, but like we talked about yesterday, if the caller is being specific by supplying transactions, thten the onus is on them to get dependencies correct.